solver/src/main/java/org/chocosolver/solver/ModelAnalyser.java

```diff
@@ -0,0 +1,1053 @@
+            List<String> list = getConstraintClassNamesOfType(cstrType);
+            for (int i = 0; i < list.size(); i++) {
+                String propType = list.get(i);
+                ConstraintTypeStatistics cstrTypeStats = createConstraintTypeStatistics(cstrType, propType);
+                ps.println(cstrTypeStats.toString(true, false, true));
+                if (i + 1 < list.size()) {
+                    ps.println();
+                }
+            }
+            ps.println();
+            ps.println();
+        }
+        ps.println("################################################");
+        ps.println("########## END OF CONSTRAINTS ANALYSIS #########");
+        ps.println("################################################");
+    }
+
+    /**
+     * Print the analysis of the <code>Model</code> in System.out.
+     *
+     * @see ModelAnalyser#printAnalysis(PrintStream)
+     */
+    public void printAnalysis() {
+        printAnalysis(System.out);
+    }
+
+    /**
+     * Print the analysis of the <code>Model</code> in the given <code>PrintStream</code>.
+     *
+     * @param ps the <code>PrintStream</code> in which to print the <code>Model</code> analysis
+     * @see java.io.PrintStream
+     */
+    public void printAnalysis(PrintStream ps) {
+        retrieveModelData();
+        ps.println();
+        ps.println("################################################################################################");
+        ps.println("#################################### BEGIN OF MODEL ANALYSIS ###################################");
+        ps.println("################################################################################################");
+        ps.println();
+        printVariableAnalysis(ps);
+        ps.println();
+        ps.println();
+        printConstraintAnalysis(ps);
+        ps.println();
+        ps.println("################################################################################################");
+        ps.println("##################################### END OF MODEL ANALYSIS ####################################");
+        ps.println("################################################################################################");
+        ps.println();
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    ////////////////////////////////////////////////     CSP GRAPH     /////////////////////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+    private static class Node<T> {
+        private static int ID = 0;
+        public final int id;
+        public final T obj;
+        public final Map<Node<T>,Integer> neighbors = new HashMap<>();
+        public Node(T t) {
+            this.id = ID++;
+            this.obj = t;
+        }
+    }
+
+    private static class Graph<T> {
```

Comment:
This is redundant with https://github.com/chocoteam/choco-solver/blob/master/solver/src/main/java/org/chocosolver/solver/trace/IOutputFactory.java#L343
which should be removed in your PR. What do you think?

Reply:
Well, Gephi output should be removed too or be preferred to GraphViz (well, Gephi can read DOT files)?
https://github.com/chocoteam/choco-solver/blob/master/solver/src/main/java/org/chocosolver/solver/trace/IOutputFactory.java#L354

Reply:
The code in ModelAnalyser is a bit redundant with the one in GraphvizGenerator, but it is necessary as the one in GraphvizGenerator is inheriting from the specificities of Searchviz.

I guess that, in order to avoid redundancy in code, we could implement a generic API to generate files for graphviz and Gephi. What do you think ? Should we keep the code as it is now, or should we implement this generic API and update the two use cases ?